File: src/test/java/ai/ancf/lmos/operator/resources/agent/ProvidedCapabilityTest.java
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.resources.agent;

import org.junit.jupiter.api.Test;

import static ai.ancf.lmos.operator.data.TestDataGenerator.createProvidedCapability;
import static org.assertj.core.api.Assertions.assertThat;

public class ProvidedCapabilityTest {

    @Test
    public void testEqualsAndHasCode(){

        var providedCapability1 = createProvidedCapability();
        var providedCapability2 = createProvidedCapability();

        assertThat(providedCapability1.equals(providedCapability2)).isTrue();
        assertThat(providedCapability1).hasSameHashCodeAs(providedCapability2);
    }

}

File: src/test/kotlin/ai/ancf/lmos/operator/resources/agent/ProvidedCapabilityTest.kt
/*
 * SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package ai.ancf.lmos.operator.resources.agent

import ai.ancf.lmos.operator.data.TestDataGenerator
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class ProvidedCapabilityTest {
    @Test
    fun testEqualsAndHasCode() {
        val providedCapability1 = TestDataGenerator.createProvidedCapability()
        val providedCapability2 = TestDataGenerator.createProvidedCapability()

        Assertions.assertThat(providedCapability1).isEqualTo(providedCapability2)
        Assertions.assertThat(providedCapability1).hasSameHashCodeAs(providedCapability2)
    }
}
